spring-cloud-scheduler-spi-test/src/main/java/org/springframework/cloud/scheduler/spi/test/EventuallyMatcher.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *          http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.scheduler.spi.test;
+
+import org.hamcrest.Description;
+import org.hamcrest.DiagnosingMatcher;
+import org.hamcrest.Matcher;
+
+/**
+ * A matcher that will evaluate another matcher repeatedly until it matches, or
+ * fail after some number of max attempts.
+ *
+ * @param <U> the type the wrapped matcher operates on
+ *
+ * @author Glenn Renfro
+ */
+public class EventuallyMatcher<U> extends DiagnosingMatcher<U> {
```

Comment:
Is this the same as https://docs.spring.io/spring-integration/api/org/springframework/integration/test/matcher/EventuallyMatcher.html?  If so, should we keep copying this around or should we get this moved into framework (or Hamcrest proper)?

Reply:
Opened issue https://github.com/hamcrest/JavaHamcrest/issues/202 with hamcrest for this.